prisma/seed.ts

```diff
@@ -0,0 +1,285 @@
+
+// POSTS - HELPER FUNCTION:
+async function createPost(userEmail: string, createType: string | CreateType) {
+  const techTags = [
+    "JavaScript",
+    "Python",
+    "Java",
+    "TypeScript",
+    "Ruby",
+    "C++",
+    "React",
+    "Angular",
+    "Vue",
+    "Django",
+    "Node.js",
+    "Express",
+    "HTML",
+    "CSS",
+    "GraphQL",
+    "REST",
+    "APIs",
+    "Docker",
+    "Kubernetes",
+    "Git",
+    "Webpack",
+    "Babel",
+    "Jenkins",
+    "Agile",
+    "Scrum",
+    "DevOps",
+    "Continuous Integration",
+    "Test-Driven Development",
+    "PostgreSQL",
+    "MongoDB",
+    "MySQL",
+    "SQLite",
+    "Redis",
+    "Firebase",
+    "AWS",
+    "Azure",
+    "Google Cloud",
+    "Encryption",
+    "OAuth",
+    "Authentication",
+    "Firewalls",
+    "UI/UX",
+    "Responsive Design",
+    "Wireframes",
+    "Prototyping",
+    "User Testing",
+    "AI",
+    "Machine Learning",
+    "AR/VR",
+    "IoT",
+  ];
+
+  const randomNumberOfArrayElements = Math.floor(Math.random() * 4) + 2;
+  const learningsOrSteps = Array.from(
+    { length: randomNumberOfArrayElements },
+    () => faker.lorem.sentence()
+  );
+
+  const randomParagraphsCount = Math.floor(Math.random() * 5) + 1;
+  const content = faker.lorem.paragraphs(randomParagraphsCount);
+
+  const randomResourcesCount = Math.floor(Math.random() * 5);
+  const resources = Array.from({ length: randomResourcesCount }, () => ({
+    label: faker.lorem.words(3),
+    link: faker.internet.url(),
+  }));
+
+  const titleAdjective = faker.hacker.adjective();
+  const titleNoun = faker.hacker.noun();
+  const capitalizedTitleAdjective = `${titleAdjective[0].toUpperCase()}${titleAdjective.slice(
+    1
+  )}`;
+  const capitalizedTitleNoun = `${titleNoun[0].toUpperCase()}${titleNoun.slice(
+    1
+  )}`;
+
+  await prisma.post.create({
+    data: {
+      title: `${capitalizedTitleAdjective} ${capitalizedTitleNoun}`,
+      createType,
+      description: faker.lorem.sentence(),
+      content,
+      ...(createType === "WORKFLOW" && { steps: learningsOrSteps }),
+      ...(createType === "KNOWLEDGE" && { learnings: learningsOrSteps }),
+      user: {
+        connect: {
+          email: userEmail,
+        },
+      },
+      tags: faker.helpers.arrayElements(techTags, { min: 1, max: 10 }),
+      ...(randomResourcesCount > 0 && {
+        resources: {
+          create: resources,
+        },
+      }),
+    },
+  });
+}
+
+async function main() {
+  try {
+    const emails: String[] = [];
+    const knowledgeLevels = [
+      "Expert in TailwindCSS",
+      "Beginner in Three.js",
+      "Advanced knowledge of MaterialUI and Shadcn",
+      "Beginner Python user",
+      "Learning AI",
+      "Proficient in React",
+      "Novice in GraphQL",
+      "Beginner in C# and .NET",
+      "Proficient in Git and Version Control",
+      "Intermediate user of Docker",
+      "Learning Cybersecurity Fundamentals",
+    ];
+    //TODO: Fix icon list
```

Comment:
This needs a bigger update in the frontend - associating each tech to its icon with the correct styles